src/dune_pkg/opam_repo.mli

```diff
@@ -0,0 +1,38 @@
+open! Stdune
+
+type t
+
+(** Create an opam repo in a local directory given by [opam_repo_dir_path]. Note
+    that [opam_repo_dir_path] is a [Filename.t] rather than, say, a
+    [Path.Outside_build_dir.t]. This is due to a current limitation in what that
+    type (and other modules in [Path]) can represent. It's desirable that the
+    [opam_repo_dir_path] argument have the following properties:
+
+    - It should be possible to build a [Repo_selection.t] from command line
+      arguments during argument parsing, which precludes the use of a [Path.t]
+      here as creating a [Path.t] requires some global state to be set up (by
+      [Common.init]) which isn't setup at the time command line arguments are
+      parsed.
+    - This argument should accept both paths within the source directory and
+      external paths, since in practical usage it will usually be pointed to
+      some out-of-source copy of the opam repository, but for unit tests it's
+      convenient that this support paths within the source directory too. This
+      rules out using [Path.External.t], [Path.Source.t], and [Path.Local.t].
+    - We want to support relative paths external to the source directory to
+      handle the case where someone has the opam repository checkoud out next to
+      their project, and runs: `dune pkg lock --opam-repository-path
+      ../opam-repository`. This rules out [Path.Outside_build_dir.t] as it
+      doesn't permit relative paths outside the source directory. *)
+val of_opam_repo_dir_path : Filename.t -> t
+
+(** Returns a path to the "packages" directory of the repo *)
+val packages_dir_path : t -> Filename.t
```

Comment:
The type of this function is weird. `Filename.t` is only for the base name of the path. I.e. it cannot contain a `/`.

This should be `string` or preferable some sort of `Path.t`

Reply:
> Filename.t is only for the base name of the path. I.e. it cannot contain a /.

This isn't true, for example `Filename.root : string` is defined to be `"/"` and the module defines a bunch of functions for working with entire paths.

That said I think you're right in that this can be a `Path.t` instead.

Reply:
Turns out this function is no longer used. Still I have updated the `Opam_repo` module to use `Path.t` instead of `Filename.t` internally.